### ISSUE TITLE
[BUG] .NET IPV4/IPV6 Issue

### ISSUE THREAD

**Issue by S74nk0** [nicehash/NiceHashMiner-Archived#397]
Opening a new issue to reference common opened issues related to SMA API failure due to .NET IP versioning error (they will be closed and referenced to this new issue to easily keep track).

### Description:
According to [FuurioBR](https://github.com/FuurioBR) NiceHash miner has the issue due to IPV4/IPV6. A workaround solution is to disable one of them more on this [here](https://github.com/nicehash/NiceHashMiner/issues/364#issuecomment-263669428).

### Solution:
Determine if IPV4 and IPV6 are present and tell NHM explicitly to use one of the versions.

### Referenced issues (closed, all new related issues should be closed and reference to this issue):
  - #364 
  - #253 
  - #327

**Comment by kenshirothefist** [comment]
Future version will get a different TCP/IP stack handler. Closing this now.

**Comment by OneOfMany07** [comment]
Closing a bug before it is fixed....  bad bad bad

**Comment by darkyat** [comment]
Can we please re-open this bug, @kenshirothefist? I just hit this bug as well in 1.7.5.13 while using IPv4 and IPv6 in parallel.

**Comment by n00bhax** [comment]
Found a way to fix it without fucking your internet by completely disabling ipv4 or ipv6:

- Go to your windows firewall settings
- Make new rules for incoming and outgoing traffic
- Choose program and allow the internet connection
- Right click on the rule you just made and go to properties 
- Now go to the tab 'protocols and ports'
- Change the type to ip6

Now will NiceHash only use ipv6 

1NhkNMtdv21Qjtx3Ee4y3yUDjceSBrax4a

**Comment by bjwood354** [comment]
I believe I was also being affected by this bug.  I had recently rebuilt my mining rig.  I am using NiceHash Miner Legacy v1.9.0.3.  Shortly after starting mining, mining would stop and the message would appear saying that mining had stopped because there is no internet connection.  I verified my internet connection was still working and it never started up automatically.  I usually have IPV6 disabled on my network adapters, but since I just recently rebuilt rig, it was still enabled along with IPV4.  I then disabled IPV6 on my network adapter and it's been working ever since making that change without interruption.  Does anyone know if this bug would impact your benchmarks if they were performed while both IPV4 and IPV6 were enabled?
